bitcoin/messages.py

```diff
@@ -335,11 +335,16 @@ def __init__(self, protover=PROTO_VERSION):
     @classmethod
     def msg_deser(cls, f, protover=PROTO_VERSION):
         c = cls()
-        c.headers = VectorSerializer.stream_deserialize(CBlock, f)
+        c.headers = VectorSerializer.stream_deserialize(CBlockHeader, f)
+        # Read 0 byte at the end
+        terminator = ser_read(f, 1)
+        assert(terminator == '\x00')
```

Comment:
The assert means that the remote end can crash the library; Bitcoin Core just ignores the byte entirely which is what you should do too.